libs/wingc/src/jsify.rs

```diff
@@ -573,46 +574,21 @@ impl<'a> JSifier<'a> {
 
 				let fqn = class_type.fqn.clone();
 
-				let scope_arg = if fqn.is_none() {
-					scope.clone()
-				} else {
-					match scope.clone() {
-						None => None,
-						Some(scope) => Some(if scope == "this" {
-							"this".to_string()
-						} else {
-							SCOPE_PARAM.to_string()
-						}),
-					}
-				};
+				let scope_arg = if fqn.is_none() { scope.clone() } else { scope.clone() };
 
 				let args = self.jsify_arg_list(&arg_list, scope_arg, id, ctx);
 
 				if let (true, Some(fqn)) = (is_preflight_class, fqn) {
-					// determine the scope to use for finding the root object
-					let node_scope = if let Some(scope) = scope {
-						scope
-					} else {
-						"this".to_string()
-					};
-
-					// if a scope is defined, use it to find the root object, otherwise use "this"
-					if node_scope != "this" {
-						new_code!(
-							expr_span,
-							format!("({SCOPE_PARAM} => {SCOPE_PARAM}.node.root.new(\""),
-							fqn,
-							"\", ",
-							ctor,
-							", ",
-							args,
-							"))(",
-							node_scope,
-							")"
-						)
-					} else {
-						new_code!(expr_span, "this.node.root.new(\"", fqn, "\", ", ctor, ", ", args, ")")
-					}
+					new_code!(
+						expr_span,
+						"globalThis.$ClassFactory.new(\"",
```

Comment:
Is it possible to hide this global call behind a static method in the platform manager?

```suggestion
						"$stdlib.platform.PlatformManager.new(\"",
```